### PR TITLE
Add/jetpack general benefits component

### PR DESCRIPTION
**Pull request by jboland88**
#### Changes proposed in this Pull Request

* This PR introduces a new component to show general benefits provided by Jetpack by plan or product. These benefits do not need to know data about a specific site.
* This PR is related to/ a sub-part of https://github.com/Automattic/wp-calypso/pull/54669 which introduces a new cancellation flow for Jetpack plans/ products

#### Testing instructions

The visual output of this component can be seen by testing PR #54669
* Either check out the branch from PR #54669 locally or use the Calypso live link on that branch to get a preview. 
* Review the testing instructions on PR #54669 to check the visual output

### CODE REVIEW

File: client/blocks/jetpack-benefits/general-benefits.tsx [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+				} ) }
+			</React.Fragment>
+		);
+	}
+
+	// Unlimited Video Hosting
+	if ( hasBusinessPlan || hasSecurityRealTimePlan || hasCompletePlan ) {
+		benefits.push(
+			<React.Fragment>
+				{ translate( 'Unlimited {{strong}}high-speed-video hosting{{/strong}}.', {
```
Comment (a8ci18n) [inline client/blocks/jetpack-benefits/general-benefits.tsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **1** times:
`translate( '{{strong}}Unlimited{{/strong}} Video hosting' )` *ES Score: 6*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/jetpack-general-benefits-component)

**Review by jboland88 — COMMENTED** [review]
Nice improvements with moving plan checks to variables and refactoring to TSX.
Seems pretty straightforward - LGTM!
I think we can update the testing plan similar to #54947 to point people to the composite branch #54947 to test the output.

@IanRamosC (not sure if you'll get pinged otherwise 😄 )

**Review by IanRamosC — APPROVED** [review]

File: client/blocks/jetpack-benefits/general-benefits.tsx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+				} ) }
+			</React.Fragment>
+		);
+	}
+
+	// Unlimited Video Hosting
+	if ( hasBusinessPlan || hasSecurityRealTimePlan || hasCompletePlan ) {
+		benefits.push(
+			<React.Fragment>
+				{ translate( 'Unlimited {{strong}}high-speed video hosting{{/strong}}.', {
```
Comment (a8ci18n) [inline client/blocks/jetpack-benefits/general-benefits.tsx:121]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **1** times:
`translate( '{{strong}}Unlimited{{/strong}} Video hosting' )` *ES Score: 6*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/jetpack-general-benefits-component)

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6455350

Hi @jboland88, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `{{strong}}Priority support{{/strong}} from Jetpack's WordPress and security experts.`
- `The ability to {{strong}}collect payments{{/strong}}.`
- `The {{strong}}Ad program{{/strong}} for WordPress.`
- `The {{strong}}Google Analytics{{/strong}} integration.`
- `Up to 13GB of {{strong}}high-speed video hosting{{/strong}}.`
- `Unlimited {{strong}}high-speed video hosting{{/strong}}.`
- `Brute force {{strong}}attack protection{{/strong}} and {{strong}}downtime monitoring{{/strong}}.`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6455350).

**Pull request merged**